prow/plugins/skip/skip.go

```diff
@@ -0,0 +1,126 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package skip implements the `/skip` command which allows users
+// to clean up commit statuses of non-blocking presubmits on PRs.
+package skip
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const pluginName = "skip"
+
+var (
+	skipRe = regexp.MustCompile(`(?mi)^/skip\s*$`)
+)
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	CreateStatus(org, repo, ref string, s github.Status) error
+	GetPullRequest(org, repo string, number int) (*github.PullRequest, error)
+	ListStatuses(org, repo, ref string) ([]github.Status, error)
+}
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	return &pluginhelp.PluginHelp{
+			Description: "The skip plugin allows users to clean up Github stale commit statuses for non-blocking jobs on a PR.",
+			WhoCanUse:   "Anyone can trigger this plugin on a PR.",
+			Usage:       "/skip",
+			Examples:    []string{"/skip"},
+		},
+		nil
+}
+
+func handleGenericComment(pc plugins.PluginClient, e github.GenericCommentEvent) error {
+	return handle(pc.GitHubClient, pc.Logger, &e, pc.Config.Presubmits[e.Repo.FullName])
+}
+
+func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent, presubmits []config.Presubmit) error {
+	if !e.IsPR || e.IssueState != "open" || e.Action != github.GenericCommentActionCreated {
+		return nil
+	}
+
+	if !skipRe.MatchString(e.Body) {
+		return nil
+	}
+
+	org := e.Repo.Owner.Login
+	repo := e.Repo.Name
+	number := e.Number
+
+	pr, err := gc.GetPullRequest(org, repo, number)
+	if err != nil {
+		resp := fmt.Sprintf("Cannot get PR #%d in %s/%s: %v", number, org, repo, err)
+		log.Warn(resp)
+		return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, resp))
+	}
+
+	statuses, err := gc.ListStatuses(org, repo, pr.Head.SHA)
+	if err != nil {
+		resp := fmt.Sprintf("Cannot get commit statuses for PR #%d in %s/%s: %v", number, org, repo, err)
+		log.Warn(resp)
+		return gc.CreateComment(org, repo, number, plugins.FormatResponseRaw(e.Body, e.HTMLURL, e.User.Login, resp))
+	}
+
+	for _, job := range presubmits {
+		// Ignore blocking jobs.
+		if job.AlwaysRun {
+			continue
+		}
+		// Ignore jobs that do not report already.
+		if job.SkipReport {
+			continue
+		}
+		// Ignore jobs that don't have a green status.
+		if !isNotSucceess(job, statuses) {
+			continue
+		}
```

Comment:
You need one additional check to ensure that the job is not a `RunIfChanged` job that is required to run.